message/spec.md

```diff
@@ -176,13 +192,38 @@ describe constraints for the metadata of messages and events, for instance the
 concrete values and patterns for the `type`, `source`, and `subject` attributes
 of a CloudEvent.
 
+Message definitions can be used in various contexts. A code generator for 
+message producers can be informed, which properties or headers have to be set, 
+and which which data types, values, or patterns are expected to produce a
+conformant message. A message consumer can use the definitions to validate
+incoming messages and to extract the metadata for routing or processing.
+
 A message group is a collection of message definitions that are related to
 each other in some application-specific way. For instance, a message group
 can be used to group all events raised by a particular application module or by
 a particular role of an application protocol exchange pattern.
 
 All message definitions MUST defined inside message groups.
 
+A message processor for a messaging or eventing channel can use a message group
+and its contained message definitions to match incoming messages to the declared
+message definitions and determine whether an incoming message is conformant to
+the expected metadata constraints. If a conformant message has been identified,
```

Comment:
```suggestion
the expected metadata constraints. If a conforming message has been identified,
```

Reply:
conformant (adjective)
(especially of technology) compatible or conforming with appropriate standards:
"vendors are using the test tool and expect to announce conformant products soon" · "using XSLT will guarantee that your output will be conformant to rules for generating XML" · "our text encoding policy is now fully conformant with the guidelines"